shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/select/projection/engine/ProjectionEngine.java

```diff
@@ -125,8 +128,16 @@ private ColumnProjection createProjection(final ColumnProjectionSegment projecti
         return new ColumnProjection(owner, projectionSegment.getColumn().getIdentifier().getValue(), projectionSegment.getAlias().orElse(null));
     }
     
-    private ExpressionProjection createProjection(final ExpressionProjectionSegment projectionSegment) {
-        return new ExpressionProjection(projectionSegment.getText(), projectionSegment.getAlias().orElse(null));
+    private ExpressionProjection createProjection(final TableSegment table, final ExpressionProjectionSegment projectionSegment) {
```

Comment:
We need add unit test and integration test to cover these new logics.

Reply:
OK, I'll do that.